src/masoniteorm/scopes/UUIDPrimaryKeyScope.py

```diff
@@ -0,0 +1,22 @@
+import uuid
+from .BaseScope import BaseScope
+
+
+class UUIDPrimaryKeyScope(BaseScope):
+    """Global scope class to use UUID4 as primary key."""
+
+    def on_boot(self, builder):
+        builder.set_global_scope(
+            "_uuid", self.set_uuid_create, action="insert"
+        )
+
+    def on_remove(self, builder):
+        pass
+
+    def set_uuid_create(self, builder):
+        builder._creates.update(
+            {
+                # TODO: later how to fetch uuid version here ?
```

Comment:
Builder._model will give you the model. This is where you can check if it has any version attributes. You can set attributes on the mixin class which will get inherited down to the model when you inherit. You can default to 4